lib/airbrake/rack/instrumentable.rb

```diff
@@ -14,15 +16,62 @@ module Rack
     # @api public
     # @since v9.2.0
     module Instrumentable
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
       def airbrake_capture_timing(method_name, label: method_name.to_s)
-        alias_method "#{method_name}_without_airbrake", method_name
+        # Check the visibility.
+        is_public = private_method_defined?(method_name)
+        is_protected = protected_method_defined?(method_name)
+        is_private = private_method_defined?(method_name)
+
+        # Make sure to forward keyword arguments too when using 2.7+ syntax.
+        use_kw_args = Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7")
+        args = use_kw_args ? "*args, **kw_args, &block" : "*args, &block"
 
-        define_method(method_name) do |*args|
-          Airbrake::Rack.capture_timing(label) do
-            __send__("#{method_name}_without_airbrake", *args)
-          end
+        # Don't chain already-prepended or operator methods.
+        self_class_idx = ancestors.index(self)
+        method_owner_idx = ancestors.index(instance_method(method_name).owner)
+        if method_owner_idx < self_class_idx || /\A\W/ =~ method_name
+          prepend(
+            Module.new do
+              module_eval <<-RUBY, __FILE__, __LINE__ + 1
+                def #{method_name}(#{args})
+                  Airbrake::Rack.capture_timing(#{label.to_s.inspect}) do
+                    super
+                  end
+                end
+              RUBY
+              # Restore the visibility.
+              protected method_name if is_protected
+              private method_name if is_private
+            end,
+          )
+        else
+          aliased = method_name.to_s.sub(/([?!=])$/, '')
```

Comment:
What's the purpose of handling punctuation? If this is just a cosmetic change, I would prefer not to mess with it.

`method_with!_without_airbrake` is clear enough because the `without_` pattern is very common in the Ruby world. By juggling punctuation we might only confuse people (even though it looks nicer).

Reply:
It's also not uncommon when using chained methods to call the wrapped methods outside of the wrapper. Not fixing the punctuation just makes that kind of thing more difficult. Also, I don't want to generate methods that break syntax when future Ruby versions might start checking that kind of thing, since it's an obvious feature that I've seen added to other languages.